services/promo.js

```diff
@@ -172,7 +172,30 @@ const middlewarePromoCatalog = async (req, res) => {
  */
 const checkPromoCatalog = async (products, user = null, lang = null, keepObject = false, populate = [], associatedProducts = false, keepPromos = false) => {
     // TODO : improve speed because it's usefull
-    if (!products || !products.length) return [];
+    if ((!products || !products.length) && (!products || !products.items || !products.items.length)) return [];
```

Comment:
I think it should be
```js
if ((!products || !products.length) || (!products || !products.items || !products.items.length)) return [];
```
can you explain why you did it that way ?

Reply:
The test is correct, we can actually have a products array as 'products' or a cart, so if products.length is undefined, maybe a products.items.legth exists,  so we return [] only if both are not defined or 0